src/legacy/core_plugins/interpreter/common/types/pointseries.ts

```diff
@@ -0,0 +1,81 @@
+/*
+ * Licensed to Elasticsearch B.V. under one or more contributor
+ * license agreements. See the NOTICE file distributed with
+ * this work for additional information regarding copyright
+ * ownership. Elasticsearch B.V. licenses this file to you under
+ * the Apache License, Version 2.0 (the "License"); you may
+ * not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+import { ExpressionType } from '../../types';
+import { Datatable } from './datatable';
+import { Render } from './render';
+
+const name = 'pointseries';
+
+/**
+ * Allowed column names in a PointSeries
+ */
+export type PointSeriesColumnName = 'x' | 'y' | 'color' | 'size' | 'text';
+
+/**
+ * Column in a PointSeries
+ */
+export interface PointSeriesColumn {
+  type: 'number' | 'string';
+  role: 'measure' | 'dimension';
+  expression: string;
+}
+
+/**
+ * Represents a collection of valid Columns in a PointSeries
+ */
+export type PointSeriesColumns = Record<PointSeriesColumnName, PointSeriesColumn> | {};
```

Comment:
Note that I added `| {}` here because from what I could tell this was a valid value which was not covered by the existing type

Reply:
I also simplified using `Record`

Reply:
I thought `Record` would allow for an empty map?

Reply:
Yeah I think you're right @clintandrewhall... I must've added `| {}` before converting to use `Record`, and then forgot to remove it. Will update, thanks for the heads up!

Reply:
Oh actually, it looks like I actually _do_ need this. Here's an example if you're curious: https://ela.st/ts-pointseries-column

 TS doesn't like the empty record in this case.